src/services/timers/index.ts

```diff
@@ -0,0 +1,86 @@
+import { prisma } from "services/db";
+import { TimersSchema } from "types/schema/timers.schema";
+import { logger } from "utils/logger";
+import { SingleTimer } from "./single-timer";
+import { gc } from "bun";
+
+export class ChannelTimer {
+  private id: number | undefined;
+  private channelName: string;
+  private fetchTimersTimer: Timer | undefined;
+  private timers: Record<number, SingleTimer> = {};
+  private send: (msg: string) => void;
+
+  constructor(channel: string, sendFn: (msg: string) => void) {
+    this.channelName = channel.replaceAll("#", "");
+    this.send = sendFn;
+
+    prisma.channel.findUnique({ where: { name: this.channelName } }).then((res) => {
+      if (!res) {
+        return;
+      }
+
+      logger.info(`Starting timers for channel: ${channel}`);
+      this.id = res.id;
+      this.syncTimersWithDB();
+      this.startFetchTimersTimer();
+    });
+  }
+
+  private async syncTimersWithDB() {
+    const timers = await prisma.timers.findMany({
+      where: {
+        channelId: this.id,
+        NOT: {
+          timeout: 0,
+        },
+      },
+    });
+
+    const parsed = TimersSchema.safeParse(timers);
```

Comment:
I think there is no need for schema here as prisma takes care of types safety etc